crates/parcel_plugin_transformer_js/src/transformer.rs

```diff
@@ -1,24 +1,894 @@
+    asset
+      .flags
+      .set(AssetFlags::SHOULD_WRAP, hoist_result.should_wrap);
+  } else {
+    if let Some(symbol_result) = result.symbol_result {
+      asset.flags |= AssetFlags::HAS_SYMBOLS;
+      asset.symbols.reserve(symbol_result.exports.len() + 1);
+      for sym in &symbol_result.exports {
+        let (local, flags) = if let Some(dep) = sym
+          .source
+          .as_ref()
+          .and_then(|source| dependency_by_specifier.get_mut(source))
+        {
+          let local = format!("${:016x}${}", dep.id(), sym.local);
+          dep.symbols.push(Symbol {
+            exported: sym.local.as_ref().into(),
+            local: local.clone(),
+            loc: Some(convert_loc(asset_file_path.clone(), &sym.loc)),
+            flags: SymbolFlags::IS_WEAK,
+          });
+          (local, SymbolFlags::IS_WEAK)
+        } else {
+          (format!("${}", sym.local).into(), SymbolFlags::empty())
+        };
+
+        asset.symbols.push(Symbol {
+          exported: sym.exported.as_ref().into(),
+          local,
+          loc: Some(convert_loc(asset_file_path.clone(), &sym.loc)),
+          flags,
+        });
+      }
+
+      for sym in symbol_result.imports {
+        if let Some(dep) = dependency_by_specifier.get_mut(&sym.source) {
+          dep
+            .symbols
+            .push(transformer_collect_imported_symbol_to_symbol(
+              &asset_file_path,
+              &sym,
+            ));
+        }
+      }
+
+      for sym in symbol_result.exports_all {
+        if let Some(dep) = dependency_by_specifier.get_mut(&sym.source) {
+          let loc = Some(convert_loc(asset_file_path.clone(), &sym.loc));
+          dep.symbols.push(make_export_all_symbol(loc));
+        }
+      }
+
+      // Add * symbol if there are CJS exports, no imports/exports at all, or the asset is wrapped.
+      // This allows accessing symbols that don't exist without errors in symbol propagation.
+      if symbol_result.has_cjs_exports
+        || (!symbol_result.is_esm
+          && asset.side_effects
+          && dependency_by_specifier.is_empty()
+          && symbol_result.exports.is_empty())
+        || (symbol_result.should_wrap
+          && !asset.symbols.as_slice().iter().any(|s| s.exported == "*"))
+      {
+        asset.symbols.push(make_export_star_symbol(asset_id));
+      }
+    } else {
+      // If the asset is wrapped, add * as a fallback
+      asset.symbols.push(make_export_star_symbol(asset_id));
+    }
+
+    // For all other imports and requires, mark everything as imported (this covers both dynamic
+    // imports and non-top-level requires.)
+    for dep in dependency_by_specifier.values_mut() {
+      if dep.symbols.is_empty() {
+        dep.symbols.push(Symbol {
+          exported: "*".into(),
+          // TODO: What does it mean?
+          local: "".into(), // format!("${}$", dep.placeholder.as_ref().unwrap_or(&dep.specifier)).into(),
```

Comment:
What is this?

Reply:
https://github.com/parcel-bundler/parcel/pull/8927

Reply:
Sorry, I mean that I don't understand what it means for `Symbol::local` to be an empty string.

Is it supposed to be like that or just in a partial state?

Reply:
oh I don't think it should be empty. shouldn't it be the commented out value? Looks like that's what I had here: https://github.com/parcel-bundler/parcel/blob/c03a29ede215967a9ec4e472a1161eed0a913af6/crates/core/src/transformers/js_transformer.rs#L866